src/backend/distributed/utils/connection_cache.c

```diff
@@ -234,9 +228,31 @@ ReportRemoteError(PGconn *connection, PGresult *result)
 		}
 	}
 
-	ereport(WARNING, (errcode(sqlState),
-					  errmsg("%s %s:%s", errorPrefix, nodeName, nodePort),
-					  errdetail("Remote message: %s", remoteMessage)));
+	/* use diffent warning messages for different error types */
+	if (sqlState == ERRCODE_CONNECTION_FAILURE)
+	{
+		ereport(WARNING, (errcode(sqlState),
+						  errmsg("connection failed to %s:%s", nodeName, nodePort),
+						  errdetail("%s", remoteMessage)));
+	}
+	else
+	{
+		if (remoteDetail == NULL)
+		{
+			ereport(WARNING, (errcode(sqlState),
+							  errmsg("remote message from %s:%s: %s", nodeName,
+									 nodePort, remoteMessage)));
+		}
+		else
+		{
+			ereport(WARNING, (errcode(sqlState),
+							  errmsg("remote message from %s:%s: %s", nodeName,
+									 nodePort, remoteMessage),
+							  errdetail("%s", remoteDetail)));
```

Comment:
Why add `errdetail` here? Why not also add e.g. `errhint`? I think this new conditional is too confusing.